test/parallel/test-querystring-maxKeys-Infinity.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+// This test was originally written to test a regression
+// that was introduced by
+// https://github.com/nodejs/node/pull/2288#issuecomment-179543894
+require('../common');
+
+const assert = require('assert');
+const parse = require('querystring').parse;
+
+// taken from express-js/body-parser
```

Comment:
This code below was borrowed from
https://github.com/expressjs/body-parser/blob/master/test/urlencoded.js#L636-L651

I'm not sure what our policy is on this. FWIW body-parser is MIT

/cc @jasnell @rvagg @mikeal